src/app/public/modules/back-to-top/back-to-top.directive.ts

```diff
@@ -38,6 +47,18 @@ import {
 })
 export class SkyBackToTopDirective implements AfterViewInit, OnDestroy {
 
+  /**
+   * Indicates if the "back to top" button should not be shown in the UI.
+   */
+  @Input()
+  public hideBackToTopButton: boolean = false;
```

Comment:
I think we have a couple options here. One would be to update `skyBackToTop` to take a value, and that value serves as an "options" object with a property called "hideButton" or something similar. That way if we ever add additional options, it can just go on that object. The other is to make the naming consistent with the other two properties, so something like `skyBackToTopButtonVisible`.

Reply:
ohhhh yeah it felt a little random adding that attribute to a `p` tag. I think I'll go with the `skyBackToTopOptions` option to not cloud up things with lots of other attributes

Reply:
You can just add `skyBackToTop` as an `@Input()` to the class and not have to add another property.

Implementation:
```
@Input()
public set skyBackToTop(value: SkyBackToTopOptions) {
  // Apply options
}
```

Usage:
```
<div [skyBackToTop]="{ hideButton: true }"></div>
```

I think I'd still keep the message stream as a distinct property, since a consumer probably won't want to create a new message stream while setting new options.